### PR TITLE
Build with CSC instead of MCS

### PR DESCRIPTION
**Pull request by chamons**
- Improve error messages from install-source when crashing

### CODE REVIEW

**Comment by chamons** [comment]
https://github.com/xamarin/xamarin-macios/pull/3246 rebased on master after PMCS landed.

File: tools/install-source/Program.cs [hunk 1]
```diff
@@ -156,13 +156,19 @@ public static int Main (string[] arguments)
...
-					Console.WriteLine($"\t{p}");
+			try {
+				var assemblySrcs = GetFilePathsFromPdb(pdbFile);
+				if (verbose) {
+					Console.WriteLine("Pdb file sources are:");
+					foreach (var p in srcs)
+						Console.WriteLine($"\t{p}");
+				}
+				srcs.UnionWith(assemblySrcs);
+			}
```
Comment (rolfbjarne) [inline tools/install-source/Program.cs]: Style: `catch` clause goes at the same line as the opening brace.

File: tools/install-source/Program.cs [hunk 2]
```diff
@@ -171,13 +177,19 @@ public static int Main (string[] arguments)
...
-					Console.WriteLine($"\t{p}");
+			try {
+				var assemblySrcs = GetFilePathsFromMdb (mdbFile);
+				if (verbose) {
+					Console.WriteLine("Mdb file sources are:");
+					foreach (var p in srcs)
+						Console.WriteLine($"\t{p}");
+				}
+				srcs.UnionWith (assemblySrcs);
+			}
```
Comment (rolfbjarne) [inline tools/install-source/Program.cs]: Style: `catch` clause goes at the same line as the opening brace.

**Comment by chamons** [comment]
So at least some of the test failures are due to the known https://github.com/mono/mono/issues/6658  that should be fixed once we bump mono.

Once we get that, we can start looking at other failures, but that one is so widespread it is likely not worth spending more time looking.

**Comment by spouliot** [comment]
@chamons it seems merging did not help :(

**Comment by chamons** [comment]
Looks like https://github.com/mono/mono/pull/7062 needs to be landed and the mono bumped (again) before we have a shot.

**Comment by spouliot** [comment]
build

**Comment by spouliot** [comment]
build

**Comment by spouliot** [comment]
Looks a lot better with 20 failures :)

```Expected file: ../MyTabbedApplication/bin/iPhoneSimulator/Debug/MyTabbedApplication.app/Xamarin.iOS.dll.mdb```

^ even more since those are easy to fix

**Comment by chamons** [comment]
build

**Comment by rolfbjarne** [comment]
build

**Comment by chamons** [comment]
Test failure was https://github.com/xamarin/maccore/issues/629

**Comment by chamons** [comment]
Others are https://github.com/xamarin/maccore/issues/647

**Pull request merged**